File: Java/day18.java
import java.util.*;

class Palindrome {
    private Deque<Character> stack;
    private Deque<Character> queue;

    public Palindrome() {
        stack = new ArrayDeque<>();
        queue = new ArrayDeque<>();
    }

    public void pushCharacter(char ch) {
        stack.addFirst(ch);
    }

    public char popCharacter() {
        return stack.removeFirst();
    }

    public void enqueueCharacter(char ch) {
        queue.addLast(ch);
    }

    public char dequeueCharacter() {
        return queue.poll();
    }
}

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // Read the string s.
        String s = sc.nextLine();

        // Create the Palindrome class object p.
        Palindrome p = new Palindrome();
        // Converting string to a character array.
        char arr[] = s.toCharArray();
        // Push all the characters of string s to stack.
        for (char c : arr) {
            p.pushCharacter(c);
        }

        // Enqueue all the characters of string s to queue.
        for (char c : arr) {
            p.enqueueCharacter(c);
        }

        boolean f = true;

        // Pop the top character from stack.
        // Dequeue the first character from queue.
        // Compare both the characters.
        for (int i = 0; i < s.length(); i++) {
            if (p.popCharacter() != p.dequeueCharacter()) {
                f = false;
                break;
            }
        }

        // Finally print whether string s is palindrome or not.
        if (f) {
            System.out.println("The word, " + s + ", is a palindrome.");
        } else {
            System.out.println("The word, " + s + ", is not a palindrome.");
        }
    }

}

File: Python/day18.py
import collections

class Palindrome:
    def __init__(self):
        self.stack = []
        self.queue = collections.deque()

    def pushCharacter(self, ch):
        self.stack.append(ch)

    def enqueueCharacter(self, ch):
        self.queue.append(ch)

    def popCharacter(self):
        return self.stack.pop()

    def dequeueCharacter(self):
        return self.queue.popleft();

# Read the string s.
s = raw_input()
# Create the Palindrome class object.
p = Palindrome()

n = len(s)
# Push all the characters of string s to stack.
for i in range(n):
    p.pushCharacter(s[i])
# Enqueue all the characters of string s to queue.
for i in range(n):
    p.enqueueCharacter(s[i])

f = True
# Pop the top character from stack.
# Dequeue the first character from queue.
# Compare both the characters.
for i in range(n):
    if p.popCharacter() != p.dequeueCharacter():
        f = False
        break

# Finally print whether string s is palindrome or not.
if f:
    sys.stdout.write ("The word, " + s + ", is a palindrome.")
else:
    sys.stdout.write ("The word, " + s + ", is not a palindrome.")
